ui/app/clinical/common/controllers/visitController.js

```diff
@@ -16,22 +16,18 @@ angular.module('bahmni.clinical')
             $scope.patientUuid = $stateParams.patientUuid;
             $scope.visitUuid = $stateParams.visitUuid;
             var tab = $stateParams.tab;
-            $scope.$on('observations.providers', function (event, observations) {
-                var consultationObservations = observations.filter(function (obs) {
-                    return obs.encounterTypeName === 'Consultation';
-                });
 
-                if (consultationObservations && consultationObservations.length > 0) {
-                    consultationObservations.sort(function (obs1, obs2) {
-                        if (obs1.observationDateTime < obs2.observationDateTime) {
-                            return -1;
-                        } else if (obs1.observationDateTime > obs2.observationDateTime) {
-                            return 1;
-                        } else {
-                            return 0;
-                        }
+            visitService.getVisit($scope.visitUuid, 'custom:(uuid,visitType,startDatetime,stopDatetime,encounters:(uuid,encounterDatetime,provider:(display),encounterType:(display)))').then(function (response) {
+                if (response.data && response.data.encounters) {
+                    var encounters = response.data.encounters;
+                    encounters = encounters.filter(function (enc) {
+                        return enc.encounterType.display == 'Consultation';
```

Comment:
Any specific reason we are hardcoding the "encounter type"? what happens if people want to include other types of encounters? I think this should be configure in app config .. and also implementation should be able to specify a list (array) of encounter types.

Reply:
@Angshu have made the changes from clinic config and updated the above code accordingly.Please verify